src/platform/windows/win_tcpconn.c

```diff
@@ -260,7 +260,13 @@ tcp_close(void *arg)
 	if (!c->closed) {
 		c->closed = true;
 		if (!nni_list_empty(&c->recv_aios)) {
-			CancelIoEx((HANDLE) c->s, &c->recv_io.olpd);
```

Comment:
So the problem is that closing the socket didn't up the waiting receivers?  Is that the situation?  (A similar problem exists with UNIX and there some unfortunate workarounds for it in the POSIX side of the code.)

Reply:
@gdamore 

FINALLY!
After your comment about closing the socket, I have checked the result of `shutdown()` and it also was raising an error (`WSAENOTCONN`).

After searching this error, it came from an issue on the call to `setsockopt(SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT)` after AcceptEx.

From the [docs](https://docs.microsoft.com/en-us/windows/win32/api/mswsock/nf-mswsock-acceptex), this function requires as a parameter the listener socket, not a boolean value.

Now it closes the connections correctly.

I have rebsaed this PR with these new changes, so it should be ready for review.

Reply:
Excellent.  Strange behavior in Windows around this, but clearly what I had was a bug.